### PR TITLE
Add capability to save audio snippets to SD card for KWS Example

### PR DESCRIPTION
**Pull request by maximreza**
KWS Example V3.1 has now the capability to write the audio snippet from the CNN memory to the SD card.
This can be useful when a new model is tested in a mobile setting without a PC (e.g. in a tradeshow when there is a lot of noise). The sound snippets are tagged with the detected keyword and written to a folder. After every power cycle or reset, a new folder is created and recorded snippets for that session saved in it.

- This feature only works for Feather board.
- To avoid SPI conflicting initialization issues if the SD card option is enabled, both wake up timer (WUT) and TFT are disabled even if they were enabled explicitly in the code.

### CODE REVIEW

**Review by aniktash — COMMENTED** [review]
just a minor item: the screenshot png of generated files includes a HEYBLIND tagged file; but heyblind is an unknown word in the general release.

File: Examples/MAX78000/CNN/kws20_demo/main.c [hunk 1]
```diff
@@ -642,6 +681,44 @@ int main(void)
...
+                if (out_class == 20)   // Unknown class
+                    LED_On(LED_RED);
+
+				int i = 0;
+				for (i=0; i<SAMPLE_SIZE; i++) {
+					// printf("%d\n",serialMicBuff[(serialMicBufIndex+i)%SAMPLE_SIZE]);
+                    snippet[i] = serialMicBuff[(serialMicBufIndex+i)%SAMPLE_SIZE];
+                }
+                snprintf(fileName, sizeof(fileName),"%04d_%s", fileCount, keywords[out_class]);
+                if (writeSoundSnippet((char*)fileName, snippetLength,  &snippet[0]) != E_NO_ERROR) {
```
Comment (Jake-Carter) [inline Examples/MAX78000/CNN/kws20_demo/main.c:709]: This always creates a file, even for low confidence detections:

image

File: Examples/MAX78000/CNN/kws20_demo/Utility/bin2wav.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    np.save(output[:-4], data)
+
+    sf.write(output, 100 * data.astype('int16'), len(data))
+
+def command_parser():
+    """
+    Return the argument parser
+    """
+    parser = argparse.ArgumentParser(description='Convert bin to audio wav')
+    parser.add_argument('-i', '--input', type=str, default = 'HEY BLIND', required = False,
```
Comment (Jake-Carter) [inline Examples/MAX78000/CNN/kws20_demo/Utility/bin2wav.py:58]: In addition to Afshin's note:  the default value here is also "HEY BLIND".  Should probably change this for general release

File: Examples/MAX78000/CNN/kws20_demo/main.c (hunk 1 above)
Comment (Jake-Carter) [reply Examples/MAX78000/CNN/kws20_demo/main.c]: Fixed in https://github.com/Analog-Devices-MSDK/msdk/pull/191/commits/4b1971a0b8aec9b09e15f15a0172ea498917f583

File: Examples/MAX78000/CNN/kws20_demo/Utility/bin2wav.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ # property whatsoever. Maxim Integrated Products, Inc. retains all
+ # ownership rights.
+ #
+ ###############################################################################
+import os
+import argparse
+import soundfile as sf
+import sounddevice as sd
+import numpy as np
+import matplotlib.pyplot as plt
```
Comment (Jake-Carter) [inline Examples/MAX78000/CNN/kws20_demo/Utility/bin2wav.py:38]: Should add a requirements.txt file for the utility scripts dependencies

**Review by Jake-Carter — CHANGES_REQUESTED** [review]
Looks good, works perfectly on my FTHR board.

Just a couple minor tweaks before merge:
* Remove references to "HEY BLIND" for this general release
* Add a requirements.txt file for the Utility folder scripts

File: Examples/MAX78000/CNN/kws20_demo/Utility/bin2wav.py (hunk 2 above)
Comment (maximreza) [reply Examples/MAX78000/CNN/kws20_demo/Utility/bin2wav.py]: removed

File: Examples/MAX78000/CNN/kws20_demo/Utility/bin2wav.py (hunk 3 above)
Comment (maximreza) [reply Examples/MAX78000/CNN/kws20_demo/Utility/bin2wav.py:38]: done

File: Examples/MAX78000/CNN/kws20_demo/main.c [hunk 4]
```diff
@@ -697,36 +697,41 @@ int main(void)
...
-                           ;
-                    }
-                    fileCount ++;
-                    LED_Off(LED_RED);
-                    LED_On(LED_GREEN);
+                	// High confidence
+                	snprintf(fileName, sizeof(fileName), "%04d_%s", fileCount, keywords[out_class]);
+                }
+                else {
+                	// Low confidence: add "L" at the end of file name
```
Comment (maximreza) [inline Examples/MAX78000/CNN/kws20_demo/main.c:719]: still want to have the low confidence detections saved. now they are marked with "_L" at the end of file name

**Review by Jake-Carter — APPROVED** [review]
Looks good, ready for merge

**Review by aniktash — APPROVED** [review]
Looks good

**Comment by Jake-Carter** [comment]
@Jake-Carter @maximreza TODO:  Disable SD by default

**Pull request merged**